olp-cpp-sdk-core/include/olp/core/network2/NetworkTypes.h

```diff
@@ -44,5 +44,29 @@ enum class RequestIdConstants : std::uint64_t {
   RequestIdMax = std::numeric_limits<RequestId>::max(),
 };
 
+/**
+ * @brief Common Network error codes.
+ */
+enum class ErrorCode {
```

Comment:
You need to give them negative numbers, else we are not able to merge them with the NetworkResponse::status_ which also includes the HTTP response codes.

Also i think the Network.h ErrorCodes are more appropriate and more user friendly.
If needed we can extend with additional error from NetworkProtocol.h ErrorCode.